src/server/src/classes/Action.ts

```diff
@@ -63,55 +50,72 @@ export class Action {
     return true;
   }
 
-  call(): boolean {
+  bet(): boolean {
+    const betAmount = this.action.betAmount as number;
+    const callAmount = this.round.highestBet - this.player.currentBet;
+
+    const mainCases = (): boolean => {
+      if (betAmount == callAmount) {
+        return this.call();
+      } else {
+        if (callAmount <= 0) {
+          console.log(`player ${this.player.id} betting amount: ${betAmount}`);
+          this.placeBet(betAmount);
+          return true;
+        } else {
+          return this.raise();
+        }
+      }
+    };
+
+    // A bit hackey but actually works really well for our needs
+    switch (true) {
+      case betAmount > this.player.chipCount:
+        return false;
+      case betAmount == this.player.chipCount:
+        return this.allIn();
+      case betAmount < callAmount:
+        return false;
+      case betAmount < 0:
+        return false;
+      case betAmount == 0:
+        return this.check();
+      case betAmount > 0:
+        return mainCases();
+      default:
```

Comment:
instead of having a bunch of cases that return false, could stack them all at the end like 

```
case betAmt < callAmt:
case [otherCases]:
default:
  return false;
```

Reply:
The thing is alot of cases rely on the higher cases to be checked first. For example, if the betAmount == 0, we cannot assume that this is a check. This is an invalid bet amount if the call amount > 0. But since this case was checked before, we can assume that by the time it gets to the betAmount == 0 case, it is a valid check. We can refactor this later, but i will switch to if statements instead of the switch.

Reply:
ah got it, sounds good